spec_parser/utils.py

```diff
@@ -162,6 +163,123 @@ def gen_rdf(self) -> None:
+                property_type = property_data["type"].split("/")[-1]
+                if property_type in vocab_dict.keys():
+                    property_dict = {
+                        "type": "string",
+                        "enum": vocab_dict[property_type]
+                    }
+                elif property_type in class_dict.keys():
+                    subclasses = self._get_all_subclasses_of(property_type, class_dict)
+                    if len(subclasses) == 1:
+                        property_dict = {
+                            "$ref": "#/definitions/" + property_type
+                        }
+                    else:
+                        property_dict = {
+                            "$ref": {
+                                "anyOf": subclasses
+                            }
+                        }
+                else:
+                    property_type = self._spdx_type_to_json_type(property_type)
+                    property_dict = {"type": property_type}
+
+                schema["definitions"][class_name]["properties"][property_name] = property_dict
+
+                if int(property_data["minCount"]) > 0:
+                    schema["definitions"][class_name]["required"].append(property_name)
+
+        schema["items"] = {
+            "anyOf": self._get_all_subclasses_of("Element", class_dict)
+        }
+
+        fname = path.join(self.args["out_dir"], f"schema.json")
+        with safe_open(fname, "w") as f:
+            json.dump(schema, f)
+
+    def _get_all_properties_from_super_classes(self, class_name, class_dict):
+        property_dict = copy(class_dict[class_name]["properties"])
+        superclass = class_dict[class_name]["SubclassOf"]
+        if superclass is None:
+            return property_dict
+        superclass = superclass.split("/")[-1]
+        if superclass in ["Payload", "xsd:string"]:
+            return property_dict
+
+        property_dict.update(self._get_all_properties_from_super_classes(superclass, class_dict))
+
+        return property_dict
+
+    def _get_all_subclasses_of(self, superclass_name, class_dict):
+        list_of_subclasses = []
+        if class_dict[superclass_name]["Instantiability"] == "Concrete":
+            list_of_subclasses.append(f"#/definitions/{superclass_name}")
+
+        for class_name, class_data in class_dict.items():
+            if class_data["SubclassOf"] == superclass_name:
+                list_of_subclasses.extend(self._get_all_subclasses_of(class_name, class_dict))
+        return list_of_subclasses
+
+
+    def _spdx_type_to_json_type(self, type_name: str) -> str:
+        if type_name in ["xsd:string", "xsd:anyURI"]:
+            return "string"
+        if type_name in ["xsd:nonNegativeInteger", "xsd:positiveInteger", "xsd:decimal"]:
+            return "number"
+        if type_name in ["xsd:boolean"]:
+            return "bool"
+
+        print("No conversion to json type implemented for " + type_name)
```

Comment:
We should probably do more than just print an error - perhaps set an error code for the CI or throw an exception.

Reply:
Currently, `context` is the (only) thing that trips this. I agree that this should raise an error, I will add that once the `context` is dealt with.

Reply:
Sorry, I meant `extension` instead of `context`.